### PR TITLE
Continuous integration for windows

### PR DESCRIPTION
**Pull request by schmave**

### CODE REVIEW

**Comment by schmave** [comment]
Thanks to the tips in PR #456 I made some modifications so that ledger builds cleanly under msys2. I configured this with Appveyor. You can see the latest build of my repo [here](https://ci.appveyor.com/project/Evan/ledger).

I discovered that the following tests are flaky on my Mac (they fail with `ctest --repeat-until-fail 50`), and fail more frequently on Windows:

26 - BaselineTest_cmd-equity (Failed)
29 - BaselineTest_cmd-pricedb (Failed)
31 - BaselineTest_cmd-prices (Failed)
100 - BaselineTest_opt-datetime-format (Failed)
142 - BaselineTest_opt-lots (Failed)
197 - BaselineTest_opt-time-report (Failed)
291 - RegressTest_786A3DD0 (Failed)

Most are due to lines in the output being in a different order than expected. Two (100 and 197) occasionally are missing an asterisk in the last line. I'd be happy to fix them, but I had a hard time navigating the code trying to figure out what determines the order. If you have a tip of where to look, I'd appreciated it.

File: src/textual.cc [hunk 1]
```diff
@@ -721,15 +721,12 @@ void instance_t::include_directive(char * line)
...
     DEBUG("textual.include", "parent file path: " << context.pathname);
-    string pathstr(context.pathname.string());
-    string::size_type pos = pathstr.rfind('/');
-    if (pos == string::npos)
-      pos = pathstr.rfind('\\');
-    if (pos != string::npos) {
-      filename = path(string(pathstr, 0, pos + 1)) / line;
-      DEBUG("textual.include", "normalized path: " << filename.string());
-    } else {
+    path parent_path = context.pathname.parent_path();
```
Comment (schmave) [inline src/textual.cc]: The old code failed for paths with mixed / and \, which happen under msys2 sometimes.

File: test/regress/383.test [hunk 2]
```diff
@@ -23,7 +23,7 @@ test bal Assets:Investments --lots --date-format %Y-%m-%d
 100 AA {2.00 GBP} [2014-01-01]  Assets:Investments
 end test
 
-test bal Assets:Investments --lots --date-format %g-%b-%d
+test bal Assets:Investments --lots --date-format %y-%b-%d
```
Comment (schmave) [inline test/regress/383.test]: %g is not supported by [Windows strftime](https://msdn.microsoft.com/en-us/library/fe06s4ak.aspx), so I switched to %y.

File: test/regress/981.test [hunk 3]
```diff
@@ -24,5 +24,6 @@ Warning: "$FILE", line 15: Metadata check failed for (Project: foo): (value =~ /
 end test
 
 test reg --limit 'payee =~ /XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/'
+__ERROR__
```
Comment (schmave) [inline test/regress/981.test]: As best I can tell, ledger has always been printing this warning message, but the harness previously didn't look at the error output unless there was expected error output. If you prefer it the old way, I'll switch it back.

**Comment by schmave** [comment]
Hi there. I'd love to get this merged. Please let me know what I can do to get this PR into shape.

I have used ledger for my personal finances and my business's finances for years now, but only because a Windows binary of v2.6 is hosted at http://www.ledger-cli.org/download.html. I am excited about making it easier for new Windows users to get started and use the latest version. Thank you for making this software!

**Comment by tbm** [comment]
I think this pull request needs to be split up.  After you made the pull request, you made 3 comments with explanations of the changes.  These should have been 3 separate pull requests with the reasons as part of the git commit log.

**Comment by jaccarmac** [comment]
My #456 contains a changeset to fix a subset of the problems this PR is trying to solve and is a little bit more robust w.r.t. `#ifdef` changes. I would also like to see a little bit more feedback about what needs to be done to get Windows fixes merged into the tree. AFAICT, none of these changes will do anything but make life easier for those trying to build or use Ledger on Windows :-).

**Comment by schmave** [comment]
OK, thanks for the feedback! I'll split this PR into smaller pieces.

**Pull request closed without merging**